src/mapping/Mapping.hpp

```diff
@@ -95,19 +113,63 @@ class Mapping {
   /// Returns true if mapping is a form of scaled consistent mapping
   bool isScaledConsistent() const;
 
+  /// Return true if the mapping is iterative an can make use of past solutions
+  bool isIterative() const;
+
+  /// Return the provided initial guess of a iterative mapping
+  const Eigen::VectorXd &initialGuess() const;
+
+  Eigen::VectorXd &initialGuess();
+
+  /// True if initialGuess().size() == 0
+  bool hasInitialGuess() const;
+
   /// Removes a computed mapping.
   virtual void clear() = 0;
 
+  /// @deprecated
+  void map(int inputDataID, int outputDataID);
+  /// @deprecated
+  void map(int inputDataID, int outputDataID, Eigen::VectorXd &initialGuess);
```

Comment:
Do we still need these in v3? I guess the goal would be to remove them before releasing. A reminder issue would help.

Reply:
Already done :smile:
https://github.com/precice/precice/issues/1672